rails-snow-queen/app/assets/javascripts/quote.js

```diff
@@ -0,0 +1,63 @@
+class Quote {
+  constructor(config) {
+    this.errors = null;
+    this.onOpen = config.onOpen;
+    
+    this.handleErrors = this.handleErrors.bind(this);
+    this.handleOpenModal = this.handleOpenModal.bind(this);
+    this.addListeners();
+  }
+  
+  addListeners() {
+    $("#submitQuoteModal").on("show.bs.modal", this.handleOpenModal);
+    document.getElementById("new_quote").addEventListener("ajax:error", this.handleErrors);
+    document.getElementById("new_quote").addEventListener("ajax:success", this.handleSuccess);
+  }
+
+  handleOpenModal() {
+    let quoteData = this.onOpen();
+
+    let splitAddress = quoteData.geocodedAddress.split(",");
+    let addressNodeInModal = document.getElementById("addressModal");
+    addressNodeInModal.innerText = splitAddress;
```

Comment:
Given that this is all _within the modal, I think the `*Modal` suffix is redundant on all of these. As a matter of fact, you might not even want the temp variables:
```javascript
document.getElementById("addressModal").innerText = splitAddress;
document.getElementById("areaModal").innerText = `${quoteData.totalAreaInSqFt.toFixed(0)}`;
// ...
```

Reply:
I did it for clarity since I have many elements with the word 'address' in multiple places, trying to be more specific with my variables